osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/ReadAlarmRegisterCommandExecutorTest.java

```diff
@@ -0,0 +1,304 @@
+                this.executor.execute(
+                    this.connectionManager, dlmsDevice, this.dto, this.messageMetadata));
+    assertThat(actual).isInstanceOf(ProtocolAdapterException.class);
+  }
+
+  @Test
+  void accessResultCodeIsNotSuccessThrowsProtocolAdapterException()
+      throws ProtocolAdapterException, IOException {
+    final DlmsDevice dlmsDevice = this.getDlmsDevice("SMR", " 5.0");
+
+    this.mockAlarmCosemObject(
+        dlmsDevice, OBIS_CODE_ALARM_REGISTER_1, DlmsObjectType.ALARM_REGISTER_1.name());
+
+    when(this.connectionManager.getDlmsMessageListener()).thenReturn(this.dlmsMessageListener);
+    when(this.connectionManager.getConnection()).thenReturn(this.dlmsConnection);
+
+    when(this.dlmsConnection.get(this.attributeAddressArgumentCaptor.capture()))
+        .thenReturn(this.getResult);
+    when(this.getResult.getResultCode()).thenReturn(AccessResultCode.TEMPORARY_FAILURE);
+
+    final Throwable actual =
+        ThrowableAssert.catchThrowable(
+            () ->
+                this.executor.execute(
+                    this.connectionManager, dlmsDevice, this.dto, this.messageMetadata));
+    assertThat(actual).isInstanceOf(ProtocolAdapterException.class);
+  }
+
+  @Test
+  void noResultThrowsProtocolAdapterException() throws ProtocolAdapterException, IOException {
+    final DlmsDevice dlmsDevice = this.getDlmsDevice("SMR", " 5.0");
+    this.mockAlarmCosemObject(
+        dlmsDevice, OBIS_CODE_ALARM_REGISTER_1, DlmsObjectType.ALARM_REGISTER_1.name());
+
+    when(this.connectionManager.getDlmsMessageListener()).thenReturn(this.dlmsMessageListener);
+    when(this.connectionManager.getConnection()).thenReturn(this.dlmsConnection);
+
+    when(this.dlmsConnection.get(this.attributeAddressArgumentCaptor.capture()))
+        .thenReturn(this.getResult);
+    when(this.getResult.getResultData()).thenReturn(null);
+    when(this.getResult.getResultCode()).thenReturn(AccessResultCode.SUCCESS);
+
+    final Throwable actual =
+        ThrowableAssert.catchThrowable(
+            () ->
+                this.executor.execute(
+                    this.connectionManager, dlmsDevice, this.dto, this.messageMetadata));
+    assertThat(actual).isInstanceOf(ProtocolAdapterException.class);
+  }
+
+  void setupAlarmRegister1(final DlmsDevice dlmsDevice) throws ProtocolAdapterException {
+
+    this.mockAlarmCosemObject(
+        dlmsDevice, OBIS_CODE_ALARM_REGISTER_1, DlmsObjectType.ALARM_REGISTER_1.name());
+
+    when(this.connectionManager.getDlmsMessageListener()).thenReturn(this.dlmsMessageListener);
+    when(this.connectionManager.getConnection()).thenReturn(this.dlmsConnection);
+  }
+
+  private void mockAlarmCosemObject(
+      final DlmsDevice dlmsDevice, final String obisCode, final String dlmsObjectTypeName)
+      throws ProtocolAdapterException {
+
+    final ObisCode newObisCode = new ObisCode(obisCode);
+
+    final AttributeAddress attributeAddress = mock(AttributeAddress.class);
```

Comment:
Is it necessary to create a mock? Suggestion:
`... = new AttributeAddress(InterfaceClass.REGISTER.id(), new ObisCode(obisCode), RegisterAttribute.VALUE.attributeId());`

Reply:
The mock is removed.